doc/farm.md

```diff
@@ -81,55 +99,52 @@ farm(exec,
     if (file) return word;
     else return {};
   },
-  [](auto x) {
-    if (x.length() > 4) std::cout << x << std::endl;
+  [](auto word) {
+    std::cout << word << " , " << word.length() << std::endl;
   },
 );
 ~~~
 ---
 
 ### Composable farm
 
-A *composable farm* has a single element:
-
-* An **Operation** performing a transformation on generated values.
-
-The input values will be generated by the upper level pattern, which will be also repsonsible for consuming the output values.
+A *composable farm* applies a **Transformer** to each data item in a stream. The
+**farm** can be inserted into another upper level pattern (which will
+be responsible for generation and consumption)
 
 ---
 **Example**
 ~~~{.cpp}
 pipeline(exec,
   stageA,
   stageB,
-  farm(exec,
-    [](auto x) {
-      if (x.length() > 4) std::cout << x << std::endl;
-    },
+  farm(exec, [](auto x) {
+    return x.length();
+  }),
   stageC
-  );
+);
 ~~~
----
 **Note**: For brevity we do not show here the details of other stages.
+---
 
 For composing complex patterns, the `farm()` function may be used to create an object that may be used later in the composition.
 
 ---
 **Example**
 ~~~{.cpp}
-auto print_long_words = farm(exec,
-    [](auto x) {
-      if (x.length() > 4) std::cout << x << std::endl;
-    }
+auto print_long_words = farm(exec, [](auto x) {
+  if (x.length() > 4) std::cout << x << std::endl;
+});
 
 pipeline(exec,
   stageA,
   stageB,
   print_long_words,
   stageC
-  );
+);
 ~~~
----
 **Note**: For brevity we do not show here the details of other stages.
```

Comment:
This note does not follow the same MD format as you used for the filter pattern.